components/elasticsearch/scripts/create.py

```diff
@@ -238,6 +238,46 @@ def _install_elasticsearch():
     utils.systemd.enable(ES_SERVICE_NAME, append_prefix=False)
 
 
+def _wait_for_shards(port, ip):
+    """
+    Function waits for activation of all available shards in Elasticsearch.
+
+    After Elasticsearch is installed and first time started there is short time
+    when shards, if created, are not started. If someone would access ES during
+    that time (e.g. by creating snapshot) he will get error
+    'SearchPhaseExecutionException[Failed to execute phase [init_scan], all
+    shards failed]'.
+
+    :param port: Elasticsearch port
+    :param ip: Ip to Elasticsearch
+    """
+
+    ctx.logger.info('Waiting for shards to be active...')
+    shards_check_timeout = 60
+    start = time.time()
+
+    url = 'http://{ip}:{port}/*/_search_shards'.format(ip=ip, port=port)
+    while True:
+        all_shards_started = True
+        try:
+            out = urllib2.urlopen(url)
+            shards = json.load(out)['shards']
+            for shard in shards:
+                all_shards_started = all_shards_started and \
+                                  (shard[0]['state'] == 'STARTED')
+        except urllib2.URLError as e:
```

Comment:
At this point, elasticsearch is necessarily up and running - function that run before `_wait_for_shards` (ie. `_configure_elasticsearch`) already send requests to it.